stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt

```diff
@@ -137,10 +138,15 @@ internal class ChannelLogic(
             logger.v { "[onQueryChannelResult] isSuccess: ${result.isSuccess}" }
             // first thing here needs to be updating configs otherwise we have a race with receiving events
             repos.insertChannelConfig(ChannelConfig(channel.type, channel.config))
-            storeStateForChannel(channel)
-        }
-            .onSuccess { channel -> channelStateLogic.propagateChannelQuery(channel, request) }
-            .onError(channelStateLogic::propagateQueryError)
+
+            if (!mutableState.insideSearch.value && !request.isFilteringAroundIdMessages()) {
+                storeStateForChannel(channel)
+            } else {
+                delay(300)
```

Comment:
This is just fixing a race condition that makes a bug in the jump to the message.